### PR TITLE
fix(jsii-project): option to disable release to npm

### PR DESCRIPTION
**Pull request by Hunter-Thompson**
`releaseToNpm: false` was not being honored by `jsii-project`, this resulted in an unwanted workflow for every project extended using `jsii-project`.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Hunter-Thompson** [comment]
This issue might be fixed by #850?

**Review by eladb — CHANGES_REQUESTED** [review]
I am not sure that's the desired behavior. By definition, jsii projects are expected to be published at least to npm and this is why this is the default behavior.

Can we leave the default to always publish to npm but also allow disablement?

**Comment by Hunter-Thompson** [comment]
@eladb, when I set `releaseToNpm` to true, the integration tests are failing. Unable to figure that out.

File: src/jsii-project.ts [hunk 1]
```diff
@@ -203,12 +203,6 @@ export class JsiiProject extends TypeScriptProject {
     }
     this.addFields({ jsii });
 
-    this.publisher?.publishToNpm({
```
Comment (eladb) [inline src/jsii-project.ts:206]: Revert this change and add an `if` statement here that will respect `options.releaseToNpm`

File: src/jsii-project.ts [hunk 2]
```diff
@@ -145,9 +145,9 @@ export class JsiiProject extends TypeScriptProject {
       // this is needed temporarily because our release workflows use the 'gh'
       // cli and its not yet available in jsii/superchain
       releaseWorkflowSetupSteps: options.releaseWorkflowSetupSteps,
-      releaseToNpm: false, // we have a jsii release workflow
```
Comment (eladb) [inline src/jsii-project.ts:148]: revert this change so that by default we pass `false` to super class

**Review by eladb — CHANGES_REQUESTED** [review]

**Review by eladb — CHANGES_REQUESTED** [review]
@Hunter-Thompson please add a unit test for this

**Pull request merged**